src/Optical/Lenses.jl

```diff
@@ -2,123 +2,374 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # See LICENSE in the project root for full license information.
 
-opticinterface(::Type{S}, insidematerial, outsidematerial, reflectance = zero(S), interfacemode = ReflectOrTransmit) where {S<:Real} = FresnelInterface{S}(insidematerial, outsidematerial, reflectance = reflectance, transmission = one(S) - reflectance, interfacemode = interfacemode)
+using .OpticSim.GlassCat: AbstractGlass, Air
+
+export SphericalLens, ConicLens, AsphericLens, FresnelLens
+
+function opticinterface(
+    ::Type{T},
+    insidematerial::AbstractGlass,
+    outsidematerial::AbstractGlass,
+    reflectance::T = zero(T),
+    interfacemode::InterfaceMode = ReflectOrTransmit
+) where {T<:Real}
+    transmission = one(T) - reflectance
+    return FresnelInterface{T}(insidematerial, outsidematerial, reflectance, transmission, interfacemode)
+end
 
 """
-    SphericalLens(insidematerial, frontvertex, frontradius, backradius, thickness, semidiameter;  lastmaterial = OpticSim.GlassCat.Air, nextmaterial = OpticSim.GlassCat.Air, frontsurfacereflectance = 0.0, backsurfacereflectance = 0.0, frontdecenter = (0, 0), backdecenter = (0, 0), interfacemode = ReflectOrTransmit)
+    SphericalLens(
+        insidematerial,
+        frontvertex,
+        frontradius,
+        backradius,
+        thickness,
+        semidiameter;
+        lastmaterial = Air,
+        nextmaterial = Air,
+        frontsurfacereflectance = 0.0,
+        backsurfacereflectance = 0.0,
+        frontdecenter = (0, 0),
+        backdecenter = (0, 0),
+        interfacemode = ReflectOrTransmit
+    )
 
 Constructs a simple cylindrical lens with spherical front and back surfaces. The side walls of the lens are absorbing.
 """
-function SphericalLens(insidematerial::T, frontvertex::S, frontradius::S, backradius::S, thickness::S, semidiameter::S; lastmaterial::Q = OpticSim.GlassCat.Air, nextmaterial::R = OpticSim.GlassCat.Air, frontsurfacereflectance::S = zero(S), backsurfacereflectance::S = zero(S), frontdecenter::Tuple{S,S} = (zero(S), zero(S)), backdecenter::Tuple{S,S} = (zero(S), zero(S)), interfacemode = ReflectOrTransmit) where {R<:OpticSim.GlassCat.AbstractGlass,Q<:OpticSim.GlassCat.AbstractGlass,T<:OpticSim.GlassCat.AbstractGlass,S<:Real}
```

Comment:
Was there any reason for parameterising `AbstractGlass` types `R`, `Q`, `T`? This was done throughout the file.

Reply:
Don't think so